src/main/java/com/iexec/blockchain/broker/BrokerService.java

```diff
@@ -66,47 +70,40 @@ public String matchOrders(BrokerOrder brokerOrder) {
         if (requestOrder == null) {
             throw new IllegalArgumentException("Request order cannot be null");
         }
-        boolean withDataset = !(requestOrder.getDataset().equals(BytesUtils.EMPTY_ADDRESS)
-                || StringUtils.isNotEmpty(requestOrder.getDataset()));
+        final boolean withDataset = withDataset(requestOrder.getDataset());
         if (datasetOrder == null && withDataset) {
             throw new IllegalArgumentException("Dataset order cannot be null");
         }
         BigInteger datasetPrice = withDataset ? datasetOrder.getDatasetprice() : BigInteger.ZERO;
         if (!hasRequesterAcceptedPrices(brokerOrder.getRequestOrder(),
                 appOrder.getAppprice(),
                 workerpoolOrder.getWorkerpoolprice(),
-                datasetPrice)) {
+                datasetPrice,
+                withDataset)) {
             throw new IllegalStateException("Incompatible prices");
         }
         //TODO check workerpool stake
-        Long deposit = iexecHubService.getChainAccount(requestOrder.getRequester())
+        long deposit = iexecHubService.getChainAccount(requestOrder.getRequester())
                 .map(ChainAccount::getDeposit)
                 .orElse(-1L);
-        if (!hasRequesterDepositedEnough(brokerOrder.getRequestOrder(), deposit)) {
+        if (!hasRequesterDepositedEnough(brokerOrder.getRequestOrder(), deposit, withDataset)) {
             throw new IllegalStateException("Deposit too low");
         }
         String beneficiary = brokerOrder.getRequestOrder().getBeneficiary();
         log.info("Matching valid orders onchain [requester:{}, beneficiary:{}, " +
                         "pool:{}, app:{}, dataset:{}]", requestOrder.getRequester(), beneficiary,
-                workerpoolOrder.getWorkerpool(), appOrder.getApp(), withDataset ? datasetOrder.getDatasetprice() : BigInteger.ZERO);
+                workerpoolOrder.getWorkerpool(), appOrder.getApp(), datasetPrice);
```

Comment:
`dataset:0xdataset`
?

Reply:
ah, I missed this. Thanks for pointing this out.

Reply:
should be fixed in 11cf4bc

Reply:
Clean thanks!